### PR TITLE
Rearrange Clone Panel

### LINKED ISSUE #23202: Clone Button Rearrangement
### Feature Description

Currently, the Clone and Download Buttons use a lot of horizontal space and only the download/clone in editor buttons are "hidden" in a Dropdown-Menu. Being used to the GitHub UI, this UI feels a bit messy and unlogical. Why does the full clone section have to be visible all the time? Why are the download buttons hidden behind "..."?

Therefore, I'd suggest rearranging the clone/download section to look similar like on GitHub, see attached Screenshots.
This would have the following benefits:
- Unless needed, the Clone/Download UI is out of the way, creating more blank space.
- Inside the dropdown menu, there is more space for the Clone/Download UI, so buttons and links can be made larger

Also, inside the dropdown, a second tab can be added (similar to how GitHub does it, see screenshot) to simplify the UI even further, especially with more items added, like PR #22378
grafik

### Screenshots

Current Implementation:
grafik

Suggested Implementation (Mockup, details may change):
grafik

### PR DESCRIPTION
**Pull request by BlenderDefender**
This pull request restores the state of the pull request #23231, unfortunately, changing the pull request head is not possible. Created as draft PR until #27931 is merged in.

Rearrange the clone panel to use less horizontal space. The following changes have been made to achieve this:
- Moved everything into the dropdown menu
- Moved the HTTPS/SSH Switch to a separate line and made the buttons larger
- Added a short description what HTTPS- and SSH-Cloning are for
- Moved the "Clone in VS Code"-Button up and added a divider to separate cloning and downloading
- Named the dropdown button "Code", added appropriate icon and made the button a primary button to raise more attention.

Before:
grafik

Current look (needs improvements):

image

This pull requests aims to fix #23202

### CODE REVIEW

**Comment by silverwind** [comment]
> Created as draft PR until https://github.com/go-gitea/gitea/pull/27931 is merged in

Is it really a dependency?

**Comment by silverwind** [comment]
I would strongly prefer if this would use tippy.js for the dropdown menu because fomantic dropdowns have numerous issues, z-index being one of them.

Take at how the PR review button is implemented, it needs a tiny bit of glue css to associate the button with the menu content.

**Comment by BlenderDefender** [comment]
Alright, thanks for the feedback 👍 
#27931 isn't a really a dependency in the sense that I could only work on this once it's merged in, but this PR almost certainly breaks the other one. If that's not a problem, I'll continue working on this and mark it ready for review as soon as I'm done.

**Comment by silverwind** [comment]
I guess it shouldn't be blocked by that semi-stale PR. It already has accumlated a number of merge conflicts.

Regarding this topic, I'm kind of neutral. On one side I like the 1-click copy behaviour of the current behaviour, but I guess we need to save space on this section which this button brings.

I suggest moving the search field to the right with this change, to the left side of the code button.

**Comment by techknowlogick** [comment]
ack. sorry for the noise on this PR. I attempted to use the auto-lintfix functionality, but it seems like git isn't accepting the reverted changes for the CSS file.

**Comment by techknowlogick** [comment]
I've just converted this to use tippy, although the styling is a bit WIP still

**Comment by techknowlogick** [comment]
This is now ready for review.

File: templates/repo/home.tmpl
Comment (lunny) [inline templates/repo/home.tmpl]: Looks like it's time to move these into a sub-template and move to a vue component in the future.

Comment (techknowlogick) [reply templates/repo/home.tmpl]: moved to a sub-template

File: templates/repo/home.tmpl
Comment (wxiaoguang) [inline templates/repo/home.tmpl]: That's wrong name (wrong abstraction). There are elements (view history) which do not belong to "clone panel"

File: web_src/js/features/repo-home.ts
Comment (wxiaoguang) [inline web_src/js/features/repo-home.ts]: You have said that https://blog.gitea.com/release-of-1.22.0/ `signaling a strategic shift as we gradually phase out jquery` but you are introducing more jquery back.

File: web_src/css/repo.css
Comment (wxiaoguang) [inline web_src/css/repo.css]: https://docs.gitea.com/contributing/guidelines-frontend :

> HTML ids and classes should use kebab-case, it's preferred to contain 2-3 feature related keywords.

File: templates/repo/clone_script.tmpl
Comment (wxiaoguang) [inline templates/repo/clone_script.tmpl]: https://docs.gitea.com/contributing/guidelines-frontend#html-attributes-and-dataset

**Comment by lafriks** [comment]
If the screenshot is up to date, the icon and text "Clone" need some spacing in between

**Comment by techknowlogick** [comment]
@lafriks That spacing has been adjusted and I'll update the screenshot shortly.

**Comment by BlenderDefender** [comment]
@techknowlogick Hey there, thanks for continuing work on this, much appreciated :)

**Comment by lunny** [comment]
Looks like something is broken. 
图片

**Comment by wxiaoguang** [comment]
@techknowlogick stale?

**Comment by wxiaoguang** [comment]
Well, no active, I will take it to 1.23

**Comment by wxiaoguang** [comment]
OK, it should be good enough now.

image

**Review by wxiaoguang — APPROVED** [review]

**Review by delvh — APPROVED** [review]

**Pull request merged**
